examples/sqlx_mysql/src/main.rs

```diff
@@ -40,7 +41,7 @@ async fn main() {
         .col(ColumnDef::new(Character::Created).date_time())
         .build(MysqlQueryBuilder);
 
-    let result = sqlx::query(&sql).execute(&mut pool).await;
+    let result = sqlx::query(&sql).execute(pool.deref_mut()).await;
```

Comment:
And use the `(&mut *pool)` syntax?

Reply:
Isn't `.deref_mut()` more explicit?

The `(&mut *pool)` seems like too much magic, especially since this is in an example

Reply:
Personally, I think `.deref_mut()` looks nicer. And, yes, as @rakshith-ravi said, less magic and more explicit.

Originally, I use `(&mut *pool)` syntax, but then I switched to `.deref_mut()`. Both ways did the same thing. See https://github.com/SeaQL/sea-orm/pull/1742/commits/ee739a02632be166f0f0a56a66cbab8f9a3c0ef2, it's SeaORM but same.

```diff
- query.execute(conn).await.map(Into::into)
+ query.execute(&mut **conn).await.map(Into::into)
```

Isn't it look bad?

With `.deref_mut()`...

```diff
- query.execute(conn).await.map(Into::into)
+ query.execute(conn.deref_mut()).await.map(Into::into)
```

Thoughts? @tyt2y3

Reply:
`&mut *pool` +1

Reply:
> The Executor impls for Transaction and PoolConnection have been deleted because they cannot exist in the new crate architecture without rewriting the Executor trait entirely.
To fix this breakage, simply add a dereference where an impl Executor is expected, as they both dereference to the inner connection type which will still implement it:
&mut transaction -> &mut *transaction
&mut connection -> &mut *connection

As per https://github.com/launchbadge/sqlx/blob/main/CHANGELOG.md#breaking